quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -554,6 +565,96 @@ impl Storage for S3CompatibleObjectStorage {
         Ok(())
     }
 
+    async fn bulk_delete<'a>(&self, paths: &[&'a Path]) -> Result<(), BulkDeleteError> {
+        let mut error = None;
+        let mut successes = Vec::with_capacity(paths.len());
+        let mut failures = HashMap::new();
+        let mut unattempted = Vec::new();
+
+        #[cfg(test)]
+        const MAX_NUM_KEYS: usize = 3;
```

Comment:
nitpick: this is probably just a matter of taste so you can ignore if you want, but I would put those declarations at the top of the file.